controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/contactinfo/ContactInfoHandler.java

```diff
@@ -0,0 +1,157 @@
+import com.yahoo.vespa.hosted.controller.restapi.ErrorResponse;
+import com.yahoo.vespa.hosted.controller.restapi.SlimeJsonResponse;
+import com.yahoo.vespa.hosted.controller.restapi.StringResponse;
+import com.yahoo.vespa.hosted.controller.tenant.AthenzTenant;
+import com.yahoo.vespa.hosted.controller.tenant.Contact;
+import com.yahoo.yolean.Exceptions;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+
+/**
+ * @author olaa
+ */
+public class ContactInfoHandler extends LoggingRequestHandler {
+
+    private final Controller controller;
+    private final Organization organization;
+
+    public ContactInfoHandler(Context ctx, Controller controller, Organization organization) {
+        super(ctx);
+        this.controller = controller;
+        this.organization = organization;
+    }
+
+    @Override
+    public HttpResponse handle(HttpRequest request) {
+        try {
+            switch (request.getMethod()) {
+                case GET:
+                    return get(request);
+                case POST:
+                    return post(request);
+                default:
+                    return ErrorResponse.methodNotAllowed("Method '" + request.getMethod() + "' is unsupported");
+            }
+        }
+        catch (IllegalArgumentException e) {
+            return ErrorResponse.badRequest(Exceptions.toMessageString(e));
+        }
+        catch (RuntimeException e) {
+            log.log(Level.WARNING, "Unexpected error handling '" + request.getUri() + "'", e);
+            return ErrorResponse.internalServerError(Exceptions.toMessageString(e));
+        }
+    }
+
+    private HttpResponse get(HttpRequest request) {
+        Path path = new Path(request.getUri().getPath());
+        if (path.matches("/contactinfo/v1/tenant/{tenant}")) return getContactInfo(path.get("tenant"), request);
+        return ErrorResponse.notFoundError("Nothing at " + path);
+    }
+
+    private HttpResponse post(HttpRequest request) {
+        Path path = new Path(request.getUri().getPath());
+        if (path.matches("/contactinfo/v1/tenant/{tenant}")) return postContactInfo(path.get("tenant"), request);
+        return ErrorResponse.notFoundError("Nothing at " + path);
+    }
+
+    private HttpResponse getContactInfo(String tenantName, HttpRequest request) {
+        Optional<AthenzTenant> tenant = controller.tenants().athenzTenant(TenantName.from(tenantName));
+        if (! tenant.isPresent()) {
+            return ErrorResponse.notFoundError("Invalid tenant " + tenantName);
+        }
+        boolean useOpsDb = getUseOpsDbFromRequest(request);
+        Optional<Contact> contact = Optional.empty();
+        if (useOpsDb) {
```

Comment:
This should just return the persisted data, so the OpsDb distinction (and injected `Organization`) can be removed. Also, we generally try to avoid using internal service names in the open source codebase.